contracts/lpp/src/lpp.rs

```diff
@@ -796,8 +858,49 @@ mod test {
         assert_eq!(withdraw, Coin::new(1110));
     }
 
-    fn coin_cw(amount: u128) -> CwCoin {
-        coin_legacy::to_cosmwasm::<TheCurrency>(amount.into())
+    #[test]
+    fn test_deposit_limit() {
+        const BORROWED: Coin<TheCurrency> = Coin::new(200);
+        const LPP_BALANCE: Coin<TheCurrency> = Coin::new(100);
+        const MIN_UTILIZATION: BoundToHundredPercent = if let Ok(percent) =
+            BoundToHundredPercent::try_from_percent(Percent::from_permille(500))
+        {
+            percent
+        } else {
+            panic!()
+        };
+        const EXPECTED_DEPOSIT_LIMIT: Coin<TheCurrency> = Coin::new(100);
+
+        let mut total: Total<TheCurrency> = Total::new();
+
+        total
+            .borrow(Timestamp::default(), BORROWED, Percent::ZERO)
+            .unwrap();
+
+        let lpp: LiquidityPool<TheCurrency> = LiquidityPool {
+            config: Config::new(
+                String::from(TheCurrency::TICKER),
+                Uint64::new(0xDEADC0DE),
+                InterestRate::new(Percent::ZERO, Percent::from_permille(500), Percent::HUNDRED)
+                    .unwrap(),
+                MIN_UTILIZATION,
+            ),
+            total,
+        };
+
+        let mock_env: Env = mock_env();
+        let mock_querier: MockQuerier =
+            MockQuerier::new(&[(mock_env.contract.address.as_str(), &[coin_cw(LPP_BALANCE)])]);
+        let mock_querier: QuerierWrapper<'_> = QuerierWrapper::new(&mock_querier);
+
+        assert_eq!(
+            lpp.deposit_limit(&mock_querier, &mock_env).unwrap(),
+            Some(EXPECTED_DEPOSIT_LIMIT)
+        );
+    }
+
+    fn coin_cw(coin: impl Into<Coin<TheCurrency>>) -> CwCoin {
```

Comment:
Kiril, please comply with all the practices we've established so far! Never ever we have used `impl T` at arg definition!